The node crashes when RPC isBlue is called

0.9.2+dev-462a686

2020-08-10|14:15:56.347 [ERROR] goroutine 6723 [running]:
github.com/Qitmeer/qitmeer/rpc.(*RpcServer).serveRequest.func1(0xc0000da4d0, 0x1af9400, 0xc0036aa960)
	/Users/baixiaoliang/Documents/GitHub/qitmeer/rpc/rpcserver.go:339 +0x123
panic(0x17cd580, 0xc009004de0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/Qitmeer/qitmeer/core/blockdag.(*Phantom).doIsBlue(0xc0001da2a0, 0x1afdb80, 0xc007f9ba00, 0x0, 0x0, 0x1446d78)
	/Users/baixiaoliang/Documents/GitHub/qitmeer/core/blockdag/phantom.go:652 +0x332
github.com/Qitmeer/qitmeer/core/blockdag.(*Phantom).IsBlue(0xc0001da2a0, 0x194a2, 0xc007f9ba00)
	/Users/baixiaoliang/Documents/GitHub/qitmeer/core/blockdag/phantom.go:642 +0x6c
github.com/Qitmeer/qitmeer/core/blockdag.(*BlockDAG).IsBlue(0xc0002be000, 0x194a2, 0x6d00)
	/Users/baixiaoliang/Documents/GitHub/qitmeer/core/blockdag/blockdag.go:1162 +0x90
github.com/Qitmeer/qitmeer/services/blkmgr.(*PublicBlockAPI).IsBlue(0xc00dd2f368, 0x3fe943f1eed10703, 0x2d245a1251b6bb52, 0x6edb9c6b64a0a7d, 0x1803ea173d728, 0x0, 0x0, 0x0, 0x0)
	/Users/baixiaoliang/Documents/GitHub/qitmeer/services/blkmgr/api.go:366 +0x120
reflect.Value.call(0xc0002ac9c0, 0xc00dd2f710, 0x13, 0x188d154, 0x4, 0xc009004db0, 0x2, 0x2, 0xc00b0534b0, 0xc0000da568, ...)
	/usr/local/go/src/reflect/value.go:460 +0x5f6
reflect.Value.Call(0xc0002ac9c0, 0xc00dd2f710, 0x13, 0xc009004db0, 0x2, 0x2, 0x1, 0x1, 0x0)
	/usr/local/go/src/reflect/value.go:321 +0xb4
github.com/Qitmeer/qitmeer/rpc.(*RpcServer).handle(0xc0000da4d0, 0x1af26e0, 0xc011372640, 0x1af9400, 0xc0036aa960, 0xc00b40e620, 0xc0036aaac0, 0x0, 0x17c0460)
	/Users/baixiaoliang/Documents/GitHub/qitmeer/rpc/rpcserver.go:606 +0x6f3
github.com/Qitmeer/qitmeer/rpc.(*RpcServer).exec(0xc0000da4d0, 0x1af26e0, 0xc011372640, 0x1af9400, 0xc0036aa960, 0xc00b40e620)
	/Users/baixiaoliang/Documents/GitHub/qitmeer/rpc/rpcserver.go:536 +0x1c4
github.com/Qitmeer/qitmeer/rpc.(*RpcServer).serveRequest(0xc0000da4d0, 0x1af27a0, 0xc009004c30, 0x1af9400, 0xc0036aa960, 0x1, 0x1, 0x0, 0x0)
	/Users/baixiaoliang/Documents/GitHub/qitmeer/rpc/rpcserver.go:399 +0x3fc
github.com/Qitmeer/qitmeer/rpc.(*RpcServer).ServeSingleRequest(...)
	/Users/baixiaoliang/Documents/GitHub/qitmeer/rpc/rpcserver.go:323
github.com/Qitmeer/qitmeer/rpc.(*RpcServer).jsonRPCRead(0xc0000da4d0, 0x1af02e0, 0xc0000fe700, 0xc0005d4600)
	/Users/baixiaoliang/Documents/GitHub/qitmeer/rpc/rpcserver.go:316 +0x508
github.com/Qitmeer/qitmeer/rpc.(*RpcServer).startHTTP.func1(0x1af02e0, 0xc0000fe700, 0xc0005d4600)
	/Users/baixiaoliang/Documents/GitHub/qitmeer/rpc/rpcserver.go:187 +0x329
net/http.HandlerFunc.ServeHTTP(0xc00d0f8c30, 0x1af02e0, 0xc0000fe700, 0xc0005d4600)
	/usr/local/go/src/net/http/server.go:2007 +0x44
net/http.(*ServeMux).ServeHTTP(0xc00f1204c0, 0x1af02e0, 0xc0000fe700, 0xc0005d4600)
	/usr/local/go/src/net/http/server.go:2387 +0x1bd
net/http.serverHandler.ServeHTTP(0xc0000feee0, 0x1af02e0, 0xc0000fe700, 0xc0005d4600)
	/usr/local/go/src/net/http/server.go:2802 +0xa4
net/http.(*conn).serve(0xc00f190460, 0x1af26e0, 0xc00e9a5e80)
	/usr/local/go/src/net/http/server.go:1890 +0x875
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:2928 +0x384

--- a/core/blockdag/phantom.go
+++ b/core/blockdag/phantom.go
@@ -649,12 +649,14 @@ func (ph *Phantom) doIsBlue(ib IBlock, fork IBlock) bool {
 	if ph.diffAnticone.Has(b.GetID()) {
 		return false
 	}
-	cur := fork.(*PhantomBlock)
-	if cur == nil {
+	var cur *PhantomBlock
+	if fork == nil {
 		cur = ph.bd.getMainFork(b, true).(*PhantomBlock)
 		if cur == nil {
 			cur = ph.getBlock(ph.mainChain.tip)
 		}
+	} else {
+		cur = fork.(*PhantomBlock)
 	}
 
 	for ; cur != nil; cur = ph.getBlock(cur.mainParent) {

